activesupport/lib/active_support/core_ext/object/json.rb

```diff
@@ -117,7 +125,18 @@ class BigDecimal
   # BigDecimal, it still has the chance to post-process the string and get the
   # real value.
   def as_json(options = nil) #:nodoc:
-    finite? ? to_s : nil
+    if
+      finite?
+      to_s
+    else
+      ActiveSupport::Deprecation.warn(<<-MSG.squish)
+        The behavior of `BigDecimal::INFINITY.as_json` and `BigDecimal::NAN.as_json`
+        returning `nil` has been deprecated, and will be removed in Rails 5.2.
```

Comment:
``` bash
$ grep -R "next version of Rails" . | wc -l
1
$ grep -R "will be removed in Rails 5.1" . | wc -l
24
```

Seems like the precedence is to say a specific version. Also it's pretty much a nailed on guarantee that we'll do a 5.2 - we've never gone from a .1 to a major release.